t/access_common_lib_terms_entitlement.t

```diff
@@ -0,0 +1,194 @@
+#!/usr/bin/perl
+
+use strict;
+use warnings;
+use Test::More;
+use File::Spec;
+
+use FindBin;
+use lib "$FindBin::Bin/lib";
+
+use RightsGlobals;
+
+use MdpConfig;
+use Auth::Auth;
+use Auth::ACL;
+use Access::Rights;
+use Database;
+use Institutions;
+use CGI;
+use Utils;
+
+use Test::File;
+
+use Data::Dumper;
+use feature qw(say);
+
+#---- MONEKYPATCHES
+no warnings 'redefine';
+
+local *Auth::Auth::auth_sys_is_SHIBBOLETH = sub {
+    return 1;
+};
+
+local *Auth::Auth::affiliation_has_emergency_access = sub {
+    return 0;
+};
+#---- MONEKYPATCHES
+
+
+my $C = new Context;
+my $cgi = new CGI;
+$C->set_object('CGI', $cgi);
+my $config = new MdpConfig(File::Spec->catdir($ENV{SDRROOT}, 'mdp-lib/Config/uber.conf'),
+                           File::Spec->catdir($ENV{SDRROOT}, 'slip-lib/Config/common.conf'));
+$C->set_object('MdpConfig', $config);
+
+my $db_user = $ENV{'MARIADB_USER'} || 'ht_testing';
+my $db = new Database($db_user);
+$C->set_object('Database', $db);
+
+my $auth = Auth::Auth->new($C);
+$C->set_object('Auth', $auth);
+
+sub setup_us_institution {
+    $ENV{REMOTE_USER} = 'user@umich.edu';
+    $ENV{eppn} = q{user@umich.edu};
+    $ENV{umichCosignFactor} = q{UMICH.EDU};
+    $ENV{Shib_Identity_Provider} = Auth::Auth::get_umich_IdP_entity_id();    
+    $ENV{affiliation} = q{};
+}
+
+sub setup_nonus_institution {
+    $ENV{REMOTE_USER} = 'user@ox.ac.edu';
+    $ENV{eppn} = q{user@ox.ac.edu};
+    delete $ENV{umichCosignFactor};
+    $ENV{Shib_Identity_Provider} = q{https://registry.shibboleth.ox.ac.uk/idp};
+    $ENV{affiliation} = q{};
+}
+
+sub setup_non_member_institution {
+    $ENV{REMOTE_USER} = 'user@gmail.com';
+    $ENV{eppn} = q{user@gmail.com};
+    $ENV{Shib_Identity_Provider} = q{gmail.com};
+    delete $ENV{umichCosignFactor};
+    $ENV{affiliation} = '';
+}
+
+sub test_attr {
+    my ( $attr, $access_profile, $location ) = @_;
+    my $id = "test.$attr\_$access_profile";
+    $ENV{TEST_GEO_IP_COUNTRY_CODE} = $location || 'US';
+
+    unless ( $attr ) {
+        print STDERR caller();
+    }
+
+    my $ar = Access::Rights->new($C, $id);
+    my $status = $ar->check_final_access_status($C, $id);
+    return $status;
+}
+
+my $num_tests = 0;
+
+my $member_tests = Test::File::load_data("$FindBin::Bin/data/access/ht_affiliate.tsv");
+my $ordinary_user_tests = Test::File::load_data("$FindBin::Bin/data/access/ordinary_user.tsv");
+
+mock_institutions($C);
+mock_acls($C);
+
+local %ENV = %ENV;
+$ENV{HTTP_HOST} = q{babel.hathitrust.org};
+$ENV{SERVER_ADDR} = q{141.213.128.185};
+$ENV{SERVER_PORT} = q{443};
+$ENV{AUTH_TYPE} = q{shibboleth};
+$ENV{entitlement} = q{urn:mace:dir:entitlement:common-lib-terms};
+
+# urn:mace:dir:entitlement:common-lib-terms from member institution
+foreach my $test ( @$member_tests ) {
```

Comment:
I'm actually not sure why this is passing right now. I'll see if I can replicate locally...

Reply:
I can replicate it passing for me on grog, but this doesn't pass for me with docker -- it reports

```
/htapps/babel/mdp-lib/t/../t/access_common_lib_terms_entitlement.t .. 1/? ASSERT_FAIL: DBI error on statement=SELECT * FROM ht_institutions WHERE entityID=?: DBD::mysql::st execute failed: Table 'ht.ht_institutions' doesn't exist at /htapps/babel/mdp-lib/t/../DbUtils.pm line 133.
```

If we want that table to exist for tests, we could use https://github.com/hathitrust/db-image instead of the SQL schema included here.

We should probably set up CI for this repository with github actions -- should be straightforward to adapt from https://github.com/hathitrust/feed/blob/master/.github/workflows/ci.yml :+1: 
```
name: Run CI

on:
  push:
    branches:
      - master

  pull_request:

jobs:
  # Run tests
  test:
    runs-on: ubuntu-latest

    steps:
      - uses: actions/checkout@v2

      - name: Check out submodules
        run: git submodule update --init metslib

      - name: Build docker image
        run: docker-compose build

      - name: Run tests
        run: docker-compose run test
```

Reply:
I'll look more tomorrow about why this is passing on grog as-is.

Reply:
Looking into this more, I think the reason this is passing is because it doesn't do anything to actually check the expected download status, and the access status doesn't differ between members and non-members.

Reply:
Indeed. Added tests for full-volume download.

Reply:
re: docker --- ah, that's interesting. I'll have to take a closer look at that. In the meanwhile, I added the non-member institution to the mocked hash which should prevent the lookup in `ht_institutions`

Reply:
:+1: Verified tests are passing in docker right now. I pushed a commit to set up CI for this.